src/test/mir-opt/simplify_try.try_identity.SimplifyLocals.after.mir

```diff
@@ -38,9 +41,14 @@ fn try_identity(_1: Result<u32, i32>) -> Result<u32, i32> {
     }
 
     bb2: {
-        ((_0 as Err).0: i32) = ((_2 as Err).0: i32); // scope 0 at $DIR/simplify_try.rs:22:13: 22:14
+        StorageLive(_4);                 // scope 0 at $DIR/simplify_try.rs:22:13: 22:14
+        StorageLive(_5);                 // scope 2 at $DIR/simplify_try.rs:22:37: 22:50
+        StorageLive(_6);                 // scope 2 at $DIR/simplify_try.rs:22:48: 22:49
+        StorageDead(_6);                 // scope 2 at $DIR/simplify_try.rs:22:49: 22:50
         Deinit(_0);                      // scope 6 at $DIR/simplify_try.rs:13:5: 13:11
         discriminant(_0) = 1;            // scope 6 at $DIR/simplify_try.rs:13:5: 13:11
+        StorageDead(_5);                 // scope 2 at $DIR/simplify_try.rs:22:50: 22:51
+        StorageDead(_4);                 // scope 0 at $DIR/simplify_try.rs:22:50: 22:51
         return;                          // scope 0 at $DIR/simplify_try.rs:26:2: 26:2
```

Comment:
Note: The destination propagation misoptimized this when optimizing `from_error`! Need to double check this is fixed in the new implementation.

Reply:
Ok, this is indeed fixed in the new implementation:

https://github.com/rust-lang/rust/blob/a758aa5aeefa43148cd84694995367f9fb9a7168/src/test/mir-opt/simplify_try.try_identity.SimplifyLocals.after.mir#L38-L41